packages/destination-actions/src/destinations/dawn/track/index.ts

```diff
@@ -0,0 +1,64 @@
+import type { ActionDefinition, RequestClient } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+import { DawnEvent } from '../dawn-types'
+
+const getEventFromPayload = (payload: Payload): DawnEvent => {
+  const event: DawnEvent = {
+    event: payload.event,
+    user_id: payload.user_id || '',
+    properties: payload.properties || {}
+  }
+  return event
+}
+
+const processData = async (request: RequestClient, settings: Settings, payload: Payload[]) => {
+  const events = payload.map((value) => getEventFromPayload(value))
+  return request('https://api.dawnai.com/segment-track', {
+    method: 'post',
+    json: events,
+    headers: {
+      authorization: `Bearer ${settings.writeKey}`
+    }
+  })
+}
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Track',
+  description: '',
+  fields: {
+    event: {
+      label: 'Event Name',
+      type: 'string',
+      description: 'The name of the action being performed.',
+      required: true,
+      default: {
+        '@path': '$.event'
+      }
+    },
+    user_id: {
+      label: 'User ID',
+      type: 'string',
+      description: 'The user ID performing the event.',
+      required: true,
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    properties: {
+      label: 'Properties',
+      type: 'object',
+      description: 'The properties of the event.',
+      required: false,
+      default: {
+        '@path': '$event' // Currently dumping the entire event object as properties
+      }
+    }
```

Comment:
```suggestion
    properties: {
      label: 'Properties',
      type: 'object',
      description: 'The properties of the event.',
      required: false,
      default: {
        '@path': '$.properties'
      }
    }
```

If you want to let the user provide the ability to map properties, then I would change the properties field to default to `'$.properties'`. 

If you want to be able to ingest the entire Segment payload, then I would either: 

1. Add a new field for the entire event: 

```
eventPayload: {
      label: 'Segment Event Payload',
      description: 'The entire Segment payload.',
      type: 'object',
      readOnly: true,
      required: true,
      default: {
        '@path': '$event'
      }
    }
```

2. Alternatively, you can access the entire event payload from the perform block by restructuring `rawData` in the second argument: 

```
  perform: async (request, { settings, payload, rawData }) => {
    return processData(request, settings, [payload])
  }
```